packages/circuit-ui/components/Table/Table.tsx

```diff
@@ -205,14 +213,21 @@ type TableState = {
  * Table interface component. It handles rendering rows/headers properly
  */
 class Table extends Component<TableProps, TableState> {
-  state: TableState = {
-    sortedRow: undefined,
-    rows: this.props.rows,
-    sortHover: undefined,
-    sortDirection: undefined,
-    scrollTop: undefined,
-    tableBodyHeight: undefined,
-  };
+  constructor(props: TableProps) {
```

Comment:
Why do we need the constructor here? Can't we just declare assign state to an object like the previous implementation?

(Sorry if I'm missing something obvious, I think my brain erased my class components knowledge after hooks came about)

Reply:
The reason was that i had to move getRows and getSortedRows above the state object because i got this bla bla bla used before definition error.

I neither wanted to move 2 methods above the state initialization nor all of them. So i decided to go the constructor way.

This looks cleaner from my perspective. If you prefer something else then i would be happy to change it.

Reply:
Ah okay, this makes sense! Thanks for the rationale here 🙌